tests/unit/amo/reducers/test_home.js

```diff
@@ -240,6 +243,73 @@ describe(__filename, () => {
       const state = homeReducer(prevState, setClientApp(CLIENT_APP_FIREFOX));
       expect(state).toEqual(initialState);
     });
+
+    it('sets `resetStateOnNextChange` to `true` after a location change on the client', () => {
+      const _config = getFakeConfig({ server: false });
+
+      const state = homeReducer(undefined, { type: LOCATION_CHANGE }, _config);
+
+      expect(state.resetStateOnNextChange).toEqual(true);
+    });
+
+    it('does not set `resetStateOnNextChange` to `true` after a location change on the server', () => {
+      const _config = getFakeConfig({ server: true });
+
+      const state = homeReducer(undefined, { type: LOCATION_CHANGE }, _config);
+
+      expect(state.resetStateOnNextChange).toEqual(false);
+    });
+
+    it('resets the state to the initial state after at least two location changes on the client', () => {
+      const _config = getFakeConfig({ server: false });
+      const { store } = dispatchClientMetadata();
+
+      _loadHomeData({
+        store,
+        collections: [
+          createFakeCollectionAddonsListResponse({
+            addons: Array(10).fill(createFakeCollectionAddon()),
+          }),
+        ],
+      });
+
+      let state = store.getState().home;
+      expect(state.collections).toHaveLength(1);
+
+      // Perform two client-side location changes.
+      state = homeReducer(state, { type: LOCATION_CHANGE }, _config);
+      state = homeReducer(state, { type: LOCATION_CHANGE }, _config);
+
+      expect(state).toEqual(initialState);
+    });
+
+    it('does not reset the state to the initial state after only one location change on the client', () => {
```

Comment:
Why is this test case needed? Isn't it already covered by the test case below?

```
     it('sets `resetStateOnNextChange` to `true` after a location change on the client', () => {
```

Reply:
This test case is testing what we actually care about, which is that the state should not be reset after the *first* location change, but only after the *second* location change. I think this is actually more important than the other two tests which check the value of `resetStateOnNextChange`. I was actually considering omitting those, because they are very "implementy" - they're really testing the internal implementation of the component and not the actual behaviour we care about.

So I don't think we should remove this test, but I would be fine with removing the other two.

WDYT?

Reply:
I think we should keep the one that has the config `server: true` at least, but the other small one can be removed.

We should leave a comment above `resetStateOnNextChange: true` to explain that when only one `LOCATION_CHANGE` has been dispatched, we only update this prop.

Reply:
If we're keeping one, we might as well keep both. It seems odd to only have a test for one side of the condition. It's not going to be much of a burden to maintain.